frontend/libs/studio-components/src/components/StudioCodelistEditor/utils/valueTypeUtils.ts

```diff
@@ -0,0 +1,56 @@
+import type { CodeList } from '../types/CodeList';
+import type { CodeListValueType } from '../types/CodeListValueType';
+import type { CodeListItemValue } from '../types/CodeListItemValue';
+
+export function updateCodeListValueType(codeList: CodeList): void {
+  const updatedCodeListValueType = getValueType(codeList);
+  coerceValues(codeList, updatedCodeListValueType);
+}
+
+export function getValueType(codeList: CodeList): CodeListValueType {
+  switch (true) {
+    case codeList.length === 0:
+      return 'undefined';
+    case someValuesAreNumbers(codeList):
+      return 'number';
+    case someValuesAreBooleans(codeList):
+      return 'boolean';
+    default:
+      return 'string';
+  }
+}
+
+function someValuesAreNumbers(codeList: CodeList): boolean {
+  return codeList.some((codeListItem) => typeof codeListItem.value === 'number');
+}
+
+function someValuesAreBooleans(codeList: CodeList): boolean {
+  return codeList.some((codeListItem) => typeof codeListItem.value === 'boolean');
+}
+
+function coerceValues(codeList: CodeList, type: CodeListValueType): void {
+  codeList.forEach((codeListItem) => {
+    switch (type) {
+      case 'number':
+        codeListItem.value = tryCoerceNumber(codeListItem.value);
+        break;
+      case 'boolean':
+        codeListItem.value = coerceBoolean(codeListItem.value);
+        break;
+      case 'string':
+      case 'undefined':
+      default:
+        codeListItem.value = String(codeListItem.value);
+    }
+  });
+}
```

Comment:
Funksjoner som brukes av React-komponenter bør være rene. Det vil si at de ikke manipulerer inndataobjektet, men returnerer et nytt objekt med modifiserte verdier. Dette gjør koden mer oversiktlig og feilsøking enklere. Når funksjonen returnerer `void` er det vanskeligere å se hva den gjør. Du kan lese mer om dette her: https://react.dev/learn/keeping-components-pure

Reply:
Takk for innspill! 
Jeg var usikker på om jeg skulle returnere `void` eller modifisere en kopi, men her er det tydeligvis en klar vinner.